CMakeLists.txt

```diff
@@ -15,60 +15,62 @@
 
-if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
-  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
-  # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-endif()
+##############################################################################
+# - build type ---------------------------------------------------------------
 
-###################################################################################################
-# - find packages we depend on --------------------------------------------------------------------
+# Set a default build type if none was specified
+rapids_cmake_build_type(Release)
 
-# Note: CUDA 10.2 debian packages are broken for cuda/std/atomic header as it is located in
-# cuda/std/std/atomic
-find_package(CUDAToolkit 10.2 REQUIRED)
+# needed for clangd and clang-tidy
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-include(cmake/Modules/CPM.cmake)
+##############################################################################
+# - compiler options ---------------------------------------------------------
 
-CPMFindPackage(
-  NAME Thrust
-  GITHUB_REPOSITORY NVIDIA/thrust
-  GIT_TAG 1.10.0
-  GIT_SHALLOW TRUE
+# * find CUDAToolkit package
+rapids_find_package(
+    CUDAToolkit REQUIRED
+    BUILD_EXPORT_SET cuco-exports
+    INSTALL_EXPORT_SET cuco-exports
 )
-thrust_create_target(Thrust)
 
-CPMAddPackage(
-    NAME libcudacxx
-    GITHUB_REPOSITORY NVIDIA/libcudacxx
-    GIT_TAG 1.3.0
-    DOWNLOAD_ONLY YES
-)
+###################################################################################################
+# - find packages we depend on --------------------------------------------------------------------
 
-# TODO: Once libcu++ exports a target, use that instead
-add_library(libcudacxx INTERFACE)
-target_include_directories(libcudacxx INTERFACE "${libcudacxx_SOURCE_DIR}/include")
+rapids_cpm_init()
 
+include(cmake/thirdparty/get_thrust.cmake)
+include(cmake/thirdparty/get_libcudacxx.cmake)
 
 ###################################################################################################
 # - cuco target   ---------------------------------------------------------------------------------
 add_library(cuco INTERFACE)
-target_include_directories(cuco INTERFACE 
-                           "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-                           $<INSTALL_INTERFACE:include>)
-target_link_libraries(cuco INTERFACE libcudacxx CUDA::cudart Thrust CUB::CUB)
+add_library(cuco::cuco ALIAS cuco)
+target_include_directories(cuco INTERFACE
+    INTERFACE   $<BUILD_INTERFACE:${CUCO_SOURCE_DIR}/include>
+                $<INSTALL_INTERFACE:include>)
+target_link_libraries(cuco INTERFACE libcudacxx CUDA::cudart cuco::Thrust)
```

Comment:
I don't know if we need to change this now but we should express a dependency on `CUDA::cudart` as it enforces the shared runtime and could cause problems if a consumer wants only the static runtime.

Reply:
If you are only trying to express a requirement on the CUDA toolkit header you should use `CUDA::toolkit`

Reply:
Done